### PR TITLE
Upgrade to React17 and bring all packages up to date

### PR DESCRIPTION
**Pull request by ZeldaZach**

### CODE REVIEW

File: webclient/tsconfig.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react"
+    "jsx": "react-jsx",
+    "noFallthroughCasesInSwitch": true
```
Comment (seavor) [inline webclient/tsconfig.json:25]: remove this.

**Review by seavor — APPROVED** [review]

**Comment by seavor** [comment]
Will merge once i'm able to test locally.

**Comment by tooomm** [comment]
Can the origin branch be deleted?

**Comment by ebbit1q** [comment]
origin is the default name for the remote, this pr's branch doesn't serve much use though

**Pull request closed without merging**